cpu/esp_common/periph/spi.c

```diff
@@ -63,6 +63,13 @@
 #define FSPI_MISO   GPIO7
 #define FSPI_MOSI   GPIO8
 
+/* DIV_UP is division which rounds up instead of down */
+#define SPI_DIV_UP(a, b)    (((a) + ((b) - 1)) / (b))
+
+/* baud_rate = 80MHz / (spi_clkdiv_pre + 1) / (spi_clkcnt_N + 1) */
+#define SPI_CLK_SRC_FREQ    80000000
```

Comment:
This frequency depends on APB clock frequencies which is 2 MHz for CPU clock of 2 MHz and 40 MHz for CPU clock of 40 MHz. The following change should work.
```suggestion
#if CONFIG_ESP32_DEFAULT_CPU_FREQ_MHZ == 2
#define SPI_CLK_SRC_FREQ    2000000
#elif CONFIG_ESP32_DEFAULT_CPU_FREQ_MHZ == 40
#define SPI_CLK_SRC_FREQ    40000000
#else
#define SPI_CLK_SRC_FREQ    80000000
#endif
```